xonsh/events.py

```diff
@@ -0,0 +1,113 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.__doc__ = "Comes with eggs"
+"""
+
+
+class Event(set):
```

Comment:
> Xontribs will load in any order. 

That is, whatever order the user decides.

> The real question is, how does amalgamate order modules?

Amalgamate orders based on import dependencies, and then lexicographically on the module name.

But all of this is getting pretty far afield.  Event needs to be an ordered container so that users can inspect the execution order and manipulate it if they want.  There are almost certainly going to be cases where one handler should only be called if a previous handler was called.  This is impossible to implement in an unordered container.